backend/firestore.rules

```diff
@@ -49,7 +49,15 @@ service cloud.firestore {
         return user.data.auth0Id == request.auth.uid;
       }
 
-      allow read: if userIsAuthenticated();
+      // These rules limit read access to the users collection by enforcing that read queries
+      // must proivide either an email address or Auth0 User ID.  This prevents the potential
+      // of a malicious client-side user from accessing every single user document at once.
+      // This works because Firestore query constraints must match the same constraints
+      // specified by the Firestore rules:
+      // https://firebase.google.com/docs/firestore/security/rules-query#secure_and_query_documents_based_on_a_field
+      allow read: if userIsAuthenticated() && resource.data.email == resource.data.email;
+      allow read: if userIsAuthenticated() && resource.data.auth0Id == resource.data.auth0Id;
+
```

Comment:
Also, these rules have not yet been pushed to production.  I tested them doing that hacky thing I do by creating my own `*_test` collections and only applying the rules to those separate collections.  I'll deploy the rules in a little bit (i.e. ~6pm EDT) once other development work slows down.  I don't want to confuse other in progress work if something is wrong with these rule updates.